js.go

```diff
@@ -148,31 +148,33 @@ func (nc *Conn) JetStream(opts ...JSOpt) (JetStream, error) {
 	return js, nil
 }
 
-// JSOpt configures options for the jetstream context.
-type JSOpt func(opts *js) error
+// JSOpt configures a JetStream context.
+type JSOpt interface {
+	configureJSContext(opts *js) error
```

Comment:
Does this mean folks can't write their own since it private?

Reply:
Since `configureJSContext(opts *js)` is private then other packages would not be able to implement the their own `JSOpt`.  To open it up, first we need to change add a public type for the options of `*js` (for example, `*JSOptions`), then  we could change this to be defined as:

```go
configureJSContext(opts *JSOptions) error
```

and we can also do something similar for `*subOpts` and have a public `*SubOptions` type.  Though since we control the interface within the same package as private and others cannot implement, we can also defer this to be done later without breaking backwards compatibility if we want and "unseal" the interface, then change it into `ConfigureJSContext(opts *JSOptions)` so that other packages can implement an option.